File: papi/lista2/exercicio3.kt
//Click on Submit and Next to continue 

fun main(args : Array<String>) {
    var numero1: Int
    var numero2: Int

    while (true) {
        try {
            print("Por favor digite o primeiro número: ") 
            val input = readLine()
            numero1 = input!!.toInt()
            break
        } catch (e : Exception) {
            println("Erro na conversão para número. Por favor tente novamente")
        }
    }

    while (true) {
        try {
            print("Por favor digite o segundo número: ") 
            val input = readLine()
            numero2 = input!!.toInt()
            break
        } catch (e : Exception) {
            println("Erro na conversão para número. Por favor tente novamente")
        }
    }

    print("A soma é: ")
    println(numero1 + numero2)
}

File: papi/lista2/exercicio3.js
// execute o npm install para instalar uma dependência externa
// estou usando o softio

const softio = require( 'softio' );
let numero1;
let numero2;

async function principal() {
    while (true) {
        numero1 = parseFloat(await softio.input( 'Por favor insira um número: ' ));

        if (numero1 > 0) {
            break;
        }

        console.log("Por favor forneça um número maior que zero");
    }

while (true) {
        numero2 = parseFloat(await softio.input( 'Por favor insira outro número: ' ));

        if (numero2 > 0) {
            break;
        }

        console.log("Por favor forneça um número maior que zero");
    }

    let soma = numero1 + numero2;
    console.log(`A soma destes números é igual a ${soma}`)
}

principal();
